caf: power_manager: Runtime configurable timeout (#6511)
nrfconnect/sdk-nrf: doc/nrf/libraries/caf/power_manager.rst

The :kconfig:`CONFIG_CAF_POWER_MANAGER_ERROR_TIMEOUT` sets the period of time after which the device is turned off upon an internal error.

> The :kconfig:`CONFIG_CAF_POWER_MANAGER_RUNTIME_TIMEOUT_SET` allows for the two aforementioned timeouts to be set during runtime.

Reviewer: My suggestion for the entire section:
```
You can configure timeout with the following Kconfig options:

* :kconfig:`CONFIG_CAF_POWER_MANAGER_TIMEOUT` - This option sets the period of time after which the application enters the low power mode.
By default, the timeout is set to 120 seconds.
* :kconfig:`CONFIG_CAF_POWER_MANAGER_ERROR_TIMEOUT` - -This option sets the period of time after which the device is turned off upon an internal error.
* :kconfig:`CONFIG_CAF_POWER_MANAGER_RUNTIME_TIMEOUT_SET` - This option allows for the other two timeout configuration options to be set during run time.
```
(We don't use "aforementioned"; `run time` is written as two words in Nordic docs)

After the reboot, the application initializes itself again.

.. |power_manager| replace:: power manager module

Update timeout intervals
========================

> Upon enabling :kconfig:`CONFIG_CAF_POWER_MANAGER_RUNTIME_TIMEOUT_SET`, the timeout interval can be set by calling (:c:func:`power_manager_configure_timeout`), and the error timeout interval can be set by calling (:c:func:`power_manager_configure_error_timeout`).
> Calling these functions will generate a :c:struct:`power_manager_configure_timeout_event` that custom modules can subscribe to.

Reviewer: ```suggestion

Upon enabling :kconfig:`CONFIG_CAF_POWER_MANAGER_RUNTIME_TIMEOUT_SET`, the timeout interval can be set by calling :c:func:`power_manager_configure_timeout`, and the error timeout interval can be set by calling :c:func:`power_manager_configure_error_timeout`.
Calling these functions generates a :c:struct:`power_manager_configure_timeout_event` that custom modules can subscribe to.
```
(Blank line needed after header)

> The event is picked up by the |power_manager| and the timeout is set immediately. That is, if the timeout is changed from 15s to 60s while the device is in the idle state, the power-down timeout will be extended by 45s for that idle interval. The following wake-up intervals will be 60s.

Reviewer: ```suggestion
The event is picked up by the |power_manager| and the timeout is set immediately.
That is, if the timeout is changed from 15s to 60s while the device is in the idle state, the power-down timeout will be extended by 45s for that idle interval.
The following wake-up intervals will be 60s.
```
(We have "one sentence per line" rule in the NCS)

> The default timeout intervals are set by :kconfig:`CONFIG_CAF_POWER_MANAGER_TIMEOUT` and :kconfig:`CONFIG_CAF_POWER_MANAGER_ERROR_TIMEOUT`.

Reviewer: ```suggestion
The default timeout intervals are set by :kconfig:`CONFIG_CAF_POWER_MANAGER_TIMEOUT` and :kconfig:`CONFIG_CAF_POWER_MANAGER_ERROR_TIMEOUT` Kconfig options.
```
